plugins/ingest/src/main/java/org/elasticsearch/ingest/IngestDocument.java

```diff
@@ -170,15 +170,15 @@ public void setPropertyValue(String path, Object value) {
     }
 
     public String getIndex() {
```

Comment:
a string is pretty lenient, but I like the idea of having a generic getter. Maybe we should have a `MetaData` enum (which has index, type, id, etc.) and then the getter would be: `getMetaData(Metadata name)`?